tests/ui/indexing_slicing_index.rs

```diff
@@ -0,0 +1,35 @@
+#![warn(clippy::indexing_slicing)]
+// We also check the out_of_bounds_indexing lint here, because it lints similar things and
+// we want to avoid false positives.
+#![warn(clippy::out_of_bounds_indexing)]
+#![allow(clippy::no_effect, clippy::unnecessary_operation)]
+
+fn main() {
+    let x = [1, 2, 3, 4];
+    let index: usize = 1;
+    x[index];
+    x[4]; // Ok, let rustc's `const_err` lint handle `usize` indexing on arrays.
+    x[1 << 3]; // Ok, let rustc's `const_err` lint handle `usize` indexing on arrays.
+
+    x[0]; // Ok, should not produce stderr.
+    x[3]; // Ok, should not produce stderr.
+
+    let y = &x;
+    y[0];
+
+    let v = vec![0; 5];
+    v[0];
+    v[10];
+    v[1 << 3];
+
+    //
+    // Continue tests at end function to minimize the changes to this file's corresponding stderr.
+    //
+
```

Comment:
```suggestion
```

This comment can be removed, I think.

Reply:
Removed via https://github.com/rust-lang/rust-clippy/pull/5130/commits/49934e70a28d599d4b46d6885c906c5541472b2d.